Detailed Instructions for Scanning Video (#420)
pupil-labs/pupil-docs: src/cloud/enrichments/README.md

- Move the glasses slowly while recording to avoid motion blur.

A good scanning video and reference image are shown below:
<div style="display:flex;" class="pb-4">
    <div style="flex-grow:1;display:flex;flex-direction:column;align-items:center;" class="pa-2">
        <iframe width="560" height="315" src="https://www.youtube.com/embed/-S5dOBqC0Uw" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
        <p>Scanning Video</p>
    </div>
    <div style="flex-grow:1;display:flex;flex-direction:column;align-items:center;" class="pa-2">
        <img src="../../media/cloud/imgs/reference_image_sample.jpg" style="padding-bottom:16px;width:60%;">
        <p>Reference Image</p>
    </div>
</div>

### Limitations
Please note that this enrichment only works for reference images showing objects that
- are static in their environment, i.e. they do not move while recording.
- are (mostly) static in their appearance, i.e. the object itself does not change while recording.

**Positive Examples**

> - Image/Painting/Poster on a wall.

Reviewer: ```suggestion
- Image affixed to a wall/surface. This could be a painting, advertisement, poster, menu, etc.
```

> - An exhibit in a musem. This could also be 3-dimensional, as long as it is static.

Reviewer: ```suggestion
- An exhibit in a museum. This could also be 3 dimensional, as long as it is static.
```

> - The cockpit of a car (or plane). While technically a car is usually moving in its environment, the interior of the car is static. We have found that this is enough to fulfill this criteria.

Reviewer: ```suggestion
- Navigation Control Center. Aircraft flight deck, automobile dashboard, machine controls. While a vehicle (car, boat, airplane, etc) is usually moving in its environment, the interior is relatively static. We have found that this _should_ be enough to fulfill this criteria.
```

Reviewer: removing "cockpit" as it is no longer a PC term in English. Gender neutral terms like "flight deck" are preferred. Additionally generalizing to include machines and control interfaces might be useful.

**Negative Examples**

> - Mobile phone/tablet. These objects typically move in their environment and show drastically changing contents on their screens.

Reviewer: ```suggestion
- Mobile phone or tablet screens. Not suitable due to dynamic movement within the environment and dynamic content displayed on screen.
```

> - Static screens (e.g. computer screen). The content of the screen usually changes dramatically. If they were showing (mostly) static content throughout all recordings, it would work.

Reviewer: ```suggestion
- Computer monitors and TV screens. The contents displayed on the monitors/screens usually changes dramatically. If the content remains static across recordings (e.g. single image) then the reference image mapper _might_ work.
```

> When in doubt if your use-case is compatible, we recommend making a quick test recording first.

Reviewer: ```suggestion
We always recommend making a quick test recording to check if your use-case is compatible.
```
